File: static/js/location.js
document.addEventListener("DOMContentLoaded", () => {
  const locationInput = document.getElementById("eventLocation");
  const statusMsg = document.getElementById("locationStatus");
  if (!locationInput || !statusMsg) return;

  const civicKeywords = [
    "park","reserve","hall","community centre","foreshore",
    "beach","oval","plaza","square","green","public","garden"
  ];

  locationInput.addEventListener("input", () => {
    const v = locationInput.value.toLowerCase();
    const matched = civicKeywords.some(k => v.includes(k));
    statusMsg.textContent = !v
      ? ""
      : matched ? "Looks like a civic/public place ✅" : "May not be a civic place ⚠️";
    statusMsg.className = "status-msg " + (matched ? "success" : "warning");
  });
});

File: modules/location.py
def validate(location_text: str) -> bool:
    """
    Simple heuristic for civic/open spaces.
    Replace with GIS lookup later.
    """
    if not location_text:
        return False
    keywords = [
        "park", "reserve", "oval", "foreshore", "beach", "hall",
        "community centre", "plaza", "square", "public", "green"
    ]
    l = location_text.lower()
    return any(k in l for k in keywords)
